src/app/data_file_manager/data_file_manager.cpp

```diff
@@ -138,6 +242,45 @@ void DataFileManager::loadAllDataForWorkers()
     DataFileManager::loadDataForReceptionists();
 }
 
+void DataFileManager::loadAllDataForVisits()
+{
+    FileManager file_manager(PATH::VISITS, "visits.proto", FileManager::FileMode::Input);
+    auto& input_file = file_manager.getFileRef();
+    clinic_data::VisitsData::Visits visit_data;
+    visit_data.ParseFromIstream(&input_file);
+    auto doctors = Clinic::getDoctors();
+    auto patients = Clinic::getPatients();
+    auto rooms = Clinic::getRooms();
+
+    for (int i = 0; i < visit_data.visits_size(); i++)
+    {
+        auto doctor = std::find_if(doctors.begin(), doctors.end(), [&](const std::shared_ptr<Doctor>& doctor) {
+            return doctor->getPesel() == visit_data.visits(i).doctor().pesel();
+        });
+        auto patient = std::find_if(patients.begin(), patients.end(), [&](const std::shared_ptr<Patient>& patient) {
+            return patient->getPesel() == visit_data.visits(i).patient().pesel();
+        });
+        auto room = std::find_if(rooms.begin(), rooms.end(), [&](const std::shared_ptr<Room>& room) {
+            return room->getRoomNumber() == visit_data.visits(i).room().room_id();
+        });
+        if (doctor != doctors.end() && patient != patients.end() && room != rooms.end())
+        {
+            Visit::createTempVisit(*doctor);
+            auto visit = Clinic::getTempVisit();
+            visit->setPatient(*patient);
+            visit->setRoom(*room);
+            visit->setVisitInformation(visit_data.visits(i).visit_information());
+            std::vector<Treatment> treatments{};
+            for (int j = 0; j < visit_data.visits(i).treatments_size(); j++)
+            {
+                treatments.push_back(toEnumTreatment(visit_data.visits(i).treatments(j)));
+            }
+            visit->updateTreatments(treatments);
```

Comment:
The body of the function got a bit long and scary. The whole if body asks for extracting into helper function (either member or in anonymous namespace), to be used like that:
```
 if (doctor != doctors.end() && patient != patients.end() && room != rooms.end())
        {
            createAndFillTempVisit(*doctor, *patient, *room);
        }
```